jobqueue/src/main/java/com/birbit/android/jobqueue/JobManagerThread.java

```diff
@@ -320,6 +363,15 @@ private void handleRunJobResult(RunJobResultMessage message) {
         }
     }
 
+    private void cancelSafely(JobHolder jobHolder, @CancelReason int cancelReason) {
+        try {
+            jobHolder.onCancel(cancelReason);
+        } catch (Throwable t) {
+            JqLog.e(t, "job's onCancel did throw an exception, ignoring...");
+        }
+        callbackManager.notifyOnCancel(jobHolder.getJob(), false);
```

Comment:
i guess it would be better to just pass the cancel reason instead of boolean. we can do it later.